gateway/reverse_proxy.go

```diff
@@ -1349,6 +1367,16 @@ func (p *ReverseProxy) WrappedServeHTTP(rw http.ResponseWriter, req *http.Reques
 
 	p.TykAPISpec.Lock()
 
+	isTimeoutEnforced, enforcedTimeout := checkHardTimeoutEnforced(p.TykAPISpec, outreq)
+
+	// limit request time with context timeout
+	if isTimeoutEnforced {
+		timeoutContext, cancel := context.WithTimeout(outreq.Context(), time.Duration(enforcedTimeout)*time.Second)
```

Comment:
@titpetric why are you using `outreq.Context` here instead or `req.Context()`?

Reply:
L1308 already modifies it; as i get it, it's used to issue the request.

Reply:
Reducing two if check to one